src/MetricsUploader/ScopedMetricTest.cpp

```diff
@@ -90,4 +92,32 @@ TEST(ScopedMetric, MoveAndSleep) {
   }
 }
 
+TEST(ScopedMetric, PauseAndResume) {
+  MockUploader uploader{};
+
+  std::chrono::milliseconds sleep_time{200};
+
+  EXPECT_CALL(uploader,
+              SendLogEvent(OrbitLogEvent::ORBIT_MAIN_WINDOW_OPEN,
+                           AllOf(Ge(sleep_time), Lt(sleep_time * 2)), OrbitLogEvent::SUCCESS))
+      .Times(2);
+
+  {
+    ScopedMetric metric{&uploader, OrbitLogEvent::ORBIT_MAIN_WINDOW_OPEN};
+    std::this_thread::sleep_for(sleep_time);
+
+    metric.Pause();
+    std::this_thread::sleep_for(sleep_time);
+    metric.Resume();
+  }
+
+  {
+    ScopedMetric metric{&uploader, OrbitLogEvent::ORBIT_MAIN_WINDOW_OPEN};
+    std::this_thread::sleep_for(sleep_time);
+
+    metric.Pause();
+    std::this_thread::sleep_for(sleep_time);
+  }
```

Comment:
I think these 2 tests might also be valuable:
1. showing that the Resume works. 
```
  {
    ScopedMetric metric{&uploader, OrbitLogEvent::ORBIT_MAIN_WINDOW_OPEN};
    std::this_thread::sleep_for(sleep_time/2);

    metric.Pause();
    std::this_thread::sleep_for(sleep_time);
    metric.Resume();
    std::this_thread::sleep_for(sleep_time/2);
  }
```

And that the move still works. Some thing like:
```
  {
    ScopedMetric metric{&uploader, OrbitLogEvent::ORBIT_MAIN_WINDOW_OPEN};
    std::this_thread::sleep_for(sleep_time/2);

    metric.Pause();
    ScopedMetric moved_metric{std::move(metric)};
    std::this_thread::sleep_for(sleep_time);
    moved_metric.Resume();
    std::this_thread::sleep_for(sleep_time/2);
  }
```

Reply:
Good points! Thanks for the suggestion!